WeaverTest/ClientGuardWrongClass.cs

```diff
@@ -0,0 +1,26 @@
+using System;
```

Comment:
Can we move this entire folder inside Mirror/WeaverTest?

it is very annoying to install plugins that pollute the root file system more than strictly required.

Reply:
You could if there was a way to tell Unity not to compile it. Is there? I felt it was safer to have it outside the assets folder, then we can build them one at a time when needed. If there's a way to tell Unity not to touch it then it can go wherever you want

Reply:
There are includePlatforms and excludePlatforms definitions that can be added to asmdefs. I'll see if we can do what you're asking that way. Otherwise I don't think it's possible. It's a requested feature here and there with no one having a concrete solution.

Reply:
If nothing else you can pass custom defines for compilation in WeaverAssembler ... each .cs could be wrapped with a define making it empty to Unity. I've found a few solutions by dumpster diving the Unity API via search box

Reply:
ohhh,  so this is outside of Assets entirely,  I missed that.

Are we supposed to run the tests with the editor test runner?   if so,  doesn't it need these classes?

Reply:
Yes from the Editor test runner.

I pass in all the appropriate assembly references from the [OneTimeSetup] of the test fixture into WeaverAssembler: UnityEngine.dll / UnityEngine,CoreModule.dll / Mirror.dll / and the textmeshpro assembly which I use to instantiate a "type from another module" error at some point. It's all based on the paths Unity is using to resolve its own compilations. I agree though it'd be tidier to have them inside a folder with the other test code. I know 100% now that it can be done with custom defines passed in from WeaverAssembler, I would just need to instrument the top and bottom of each test with an #if / #endif and Unity will see them as blank sources. This is where a preprocessor #include statement would be nice